addons/sale_crm/models/crm_team.py

```diff
@@ -42,3 +42,52 @@ def _extra_sql_conditions(self):
         if self.use_opportunities and self._context.get('in_sales_app'):
             return "AND state in ('sale', 'done', 'pos_done')"
         return super(CrmTeam,self)._extra_sql_conditions()
+
+    def _merge_leads(self, leads, values):
+        """ Remove opportunities or leads from ["leads_dups_dict"] that
+        have sale orders and the company of the sale order is not the same
+        as the company of the head lead, its sale order, or current company.
+
+        :param leads: recordset of leads to assign to current team;
+        :param values: dictionary in the following form
+                       {
+                            'lead_duplicates': crm.lead(),
+                            'leads_assigned': crm.lead(),
+                            'leads_dups_dict': {crm.lead(): crm.lead()}
+                       };
+        :return: Opportunities or leads that have sale orders, and the company
+        of the sale order is not the same as the company of the head lead, its
+        sale order, or current company.
+        (
+            {'leads_with_different_company': opportunities}
+        )
+        """
+        opportunities = values['lead_duplicates']._sort_by_confidence_level(reverse=True)
+        if opportunities.order_ids \
+           and (len(opportunities.order_ids.company_id) > 1 or len(opportunities.company_id) > 1):
+            is_return = False
+            head_opportunity = opportunities[0]
+            current_company = head_opportunity.company_id or head_opportunity.order_ids.company_id or self.env.company
+            leads_with_so_different_company = opportunities.filtered(
+                lambda lead: lead.order_ids and lead.order_ids.company_id != current_company
+            )
+            if len(opportunities - leads_with_so_different_company) < 2:
+                is_return = True
+            else:
+                if leads in leads_with_so_different_company:
```

Comment:
I don't think this, or the dict update, works if we input multiple leads.

Reply:
Hello @thiryr 
In leads, there is only one lead (candidate lead). see https://github.com/odoo/odoo/blob/15.0/addons/crm/models/crm_team.py#L506-L507.

I don't think a loop is required for leads in the below line of codes:
https://github.com/odoo/odoo/blob/15.0/addons/crm/models/crm_team.py#L576
https://github.com/odoo/odoo/blob/15.0/addons/crm/models/crm_team.py#L556

What do you think?

Reply:
Yes I aggre, it doesn't seem to ever be used. But since this is stable, it's possible someone wrote an addon using these methods with the expectation that they can handle multiple leads.

I wouldn't remove support for it, at least not in stable.